### PR TITLE
CLN: reorg pandas.tseries

### PR DESCRIPTION
**Pull request by jreback**
xref #13634

### CODE REVIEW

**Comment by jreback** [comment]
this finishes the api cleanup and moves everything but:

- ``freqencies``, ``holidays``, ``offsets``
- ``util``, and ``plotting`` (which are deprecated)

from ``pandas.tseries``.

all of the indexes code is now actually in ``pandas.core.indexes`` (tests were moved previously).

File: doc/source/conf.py [hunk 1]
```diff
@@ -231,7 +231,7 @@
 moved_api_pages = [
     'pandas.core.common.isnull', 'pandas.core.common.notnull', 'pandas.core.reshape.get_dummies',
     'pandas.tools.merge.concat', 'pandas.tools.merge.merge', 'pandas.tools.pivot.pivot_table',
-    'pandas.tseries.tools.to_datetime', 'pandas.io.clipboard.read_clipboard', 'pandas.io.excel.ExcelFile.parse',
```
Comment (jorisvandenbossche) [inline doc/source/conf.py]: this shouldn't be changed

**Review by jorisvandenbossche — COMMENTED** [review]
We also need to think of something regarding the API docs. Currently that uses `pandas.tseries.resample` (eg http://pandas.pydata.org/pandas-docs/stable/generated/pandas.tseries.resample.Resampler.aggregate.html)

That will have to change, meaning all doc links won't work anymore (not that big of a problem I think), but maybe we should think about a more long term option, and also expose this somewhere less nested that will not change? (but not sure where, as I would not add it top-level)

File: doc/source/conf.py (hunk 1 above)
Comment (jreback) [inline doc/source/conf.py]: fixed

**Comment by jreback** [comment]
> We also need to think of something regarding the API docs. Currently that uses pandas.tseries.resample (eg http://pandas.pydata.org/pandas-docs/stable/generated/pandas.tseries.resample.Resampler.aggregate.html)

this just changes to ``pandas.core.resample`` so I don't see a problem.

**Comment by jorisvandenbossche** [comment]
Yes, but that means that *all* our online pages change url

**Comment by jorisvandenbossche** [comment]
You created here the `core.tools` module, which seems like a good idea, but then it may make sense to also put `to_numeric` there for consistency?

**Comment by jreback** [comment]
> You created here the core.tools module, which seems like a good idea, but then it may make sense to also put to_numeric there for consistency?

yeah was debating this. ok will move it.

> Yes, but that means that all our online pages change url

and? *lots* of thinks moved.

**Comment by jorisvandenbossche** [comment]
> and? lots of thinks moved.

yes, but up to now (I mean the reorg the last days), no 'public' things. And `Resampler` is not really public, apart from that it is used in the docs.

**Comment by jreback** [comment]
can we do a re-direct?

**Comment by jorisvandenbossche** [comment]
> can we do a re-direct?

yes, that is certainly possible. 

My concern is only that the urls currently depend a bit on the 'implementation details' in `core` (as one of the reasons of this reorg and the more explicit saying that `core` is private, is to be able to refactor internally). So just wondering whether it would make sense to expose eg GroupBy, Rolling, Resampler in eg `pandas.core` directly, not as a public API, as no user should directly use those objects, but to have a stable doc url independent of where or how in core it is implemented.

Although chance is high that the `pandas.core.groupby.GroupBy`, `pandas.core.window.Rolling`, `pandas.core.resample.Resampler` paths will not directly be changed again, so maybe not that important.

**Comment by jreback** [comment]
could u do some kind ornate shim for the actual links?

maybe in conf.py make a namespace like

pandas.doclinks or something then just assign the classes there ?

**Comment by jorisvandenbossche** [comment]
I am not sure something like that would work. As sphinx has to actually import it to get the docstring (I think, not sure how it works under the hood)

**Pull request merged**